test/runtime/Policies.go

```diff
@@ -486,6 +487,62 @@ var _ = Describe("RuntimeValidatedPolicies", func() {
 		connectivityTest(allRequests, helpers.App2, helpers.Httpd1, true)
 	})
 
+	It("Tests Endpoint Connectivity Functions After Daemon Configuration Is Updated", func() {
+		httpd1DockerNetworking, err := vm.ContainerInspectNet(helpers.Httpd1)
+		Expect(err).ToNot(HaveOccurred(), "unable to get container networking metadata for %s", helpers.Httpd1)
+
+		By("Importing policy and waiting for revision to increase for endpoints")
+		_, err = vm.PolicyImportAndWait(vm.GetFullPath(policiesL7JSON), helpers.HelperTimeout)
+		Expect(err).ToNot(HaveOccurred(), "unable to import policy after timeout")
+
+		By("Trying to access %s:80/public from %s before daemon configuration is updated (should be allowed by policy)", helpers.Httpd1, helpers.App1)
+		res := vm.ContainerExec(helpers.App1, helpers.CurlFail("http://%s:80/public", httpd1DockerNetworking[helpers.IPv4]))
+		res.ExpectSuccess("unable to access %s:80/public from %s (should have worked)", helpers.Httpd1, helpers.App1)
+
+		currentRev, err := vm.PolicyGetRevision()
+		Expect(err).ToNot(HaveOccurred(), "unable to get policy revision")
+
+		By("Getting configuration for daemon")
+		daemonDebugConfig, err := vm.ExecCilium("config -o json").Filter("{.Debug}")
+		Expect(err).ToNot(HaveOccurred(), "Unable to get configuration for daemon")
+
+		daemonDebugConfigString := daemonDebugConfig.String()
+
+		var daemonDebugConfigSwitched string
+
+		switch daemonDebugConfigString {
+		case "Disabled":
+			daemonDebugConfigSwitched = "Enabled"
+		case "Enabled":
+			daemonDebugConfigSwitched = "Disabled"
+		default:
+			ginkgo.Fail(fmt.Sprintf("invalid configuration value for daemon: Debug=%s", daemonDebugConfigString))
+		}
+
+		// TODO: would be a good idea to factor out daemon configuration updates
+		// into a function in the future.
+		By("Changing daemon configuration from Debug=%s to Debug=%s to induce policy recalculation for endpoints", daemonDebugConfigString, daemonDebugConfigSwitched)
+		res = vm.ExecCilium(fmt.Sprintf("config Debug=%s", daemonDebugConfigSwitched))
+		res.ExpectSuccess("unable to change daemon configuration")
+
+		By("Waiting for policy revision to increase after daemon configuration change")
+		res = vm.PolicyWait(currentRev + 1)
```

Comment:
Here I would add and assert that validates that the current PolicyRevision is equal to currentRev +1. What happens if we trigger two times the PolicyUpdate and we wait to the version before to the current one. An assert that validates the version will be great.

Reply:
What do you mean by 'if we trigger two times the PolicyUpdate'? Are you saying that it could be triggered between when we apply the configuration and then wait by some other call to the function within Cilium? This could happen, but I don't really see another way around this, because we have to wait for a specific revision. The policy revision could change after the assertion is made by one of these asynchronous calls to `TriggerPolicyUpdates`.

Reply:
```
		By("Waiting for policy revision to increase after daemon configuration change")

		version, err := vm.PolicyGetRevision()
		Expect(err).To(BeNil())
		Expect(version).To(BeNumerically(">=", currentRevision+1))

		res = vm.PolicyWait(version)
		res.ExpectSuccess("policy revision was not bumped after daemon configuration changes")

		By("Changing daemon configuration back from Debug=%s to Debug=%s", daemonDebugConfigSwitched, daemonDebugConfigString)

		res = vm.ExecCilium(fmt.Sprintf("config Debug=%s", daemonDebugConfigString))
		res.ExpectSuccess("unable to change daemon configuration")

		By("Waiting for policy revision to increase after daemon configuration change")

		currentRev, err := vm.PolicyGetRevision()
		Expect(err).To(BeNil())
		Expect(version).To(BeNumerically(">=", version+1))
		res = vm.PolicyWait(currentRev)

```

With this code, you always make sure that if something async trigger it, the version that you get, and after the debug is always the newer one. does this make sense to you? 

Regards

Reply:
`cilium policy wait <revision number>` waits until the desired revision number has been reached, or exits if it already has been reached (i.e., <policy repository revision> >= <revision number>), so that's basically performing the check you desire with : 

```
version, err := vm.PolicyGetRevision()
Expect(err).To(BeNil())
Expect(version).To(BeNumerically(">=", currentRevision+1))
```

Also, calling `PolicyWait` after calling `PolicyGetRevision` will always succeed given the above explanation, because the call to `TriggerPolicyUpdates(true)` which is invoked upon configuration updates for the daemon synchronously bumps the policy repository revision. So, these calls in your example (and my code in this PR) will always succeed, even if the endpoints themselves haven't been re-built yet. There needs to be a helper with checks that the policy revision number for each *endpoint* is bumped to a certain value, because per-endpoint, that value is only set if the build for endpoints has succeeded.

I'll work on adding such a helper, as it will be quite useful instead of checking endpoint state, which is prone to race-y behavior.

Reply:
Sigh, actually, I'm wrong. `policy wait` waits for all endpoints to be at the given policy revision, so the new helper isn't needed. I now get what you are saying, and will implement it.

Reply:
Fixed.